### PR TITLE
Edit extensionAttributes

### PR DESCRIPTION
**Pull request by chasen-bettinger**
Altering the extensionAttributes properties on the `user_endpoint`
entity to prevent the entity size from growing too large in the event
that every `extensionAttribute` property is added onto the entity.

# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Type of change

Please leave any irrelevant options unchecked.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [x] This change requires a documentation update

### General Development Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [x] I have updated the `CHANGELOG.md` file to describe my changes
- [x] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/steps/devices/__snapshots__/converters.test.ts.snap [hunk 1]
```diff
@@ -345,13 +352,8 @@ Object {
...
   "encrypted": true,
   "enrolledOn": 1591217314697,
-  "extensionAttribute.123": Array [
-    "foo123",
-    "bar123",
-  ],
-  "extensionAttribute.456": Array [
-    "foo456",
-    "bar456",
+  "extensionAttribute.Deployment status": Array [
```
Comment (aiwilliams) [inline src/steps/devices/__snapshots__/converters.test.ts.snap:355]: Whoohoo! Gotta love these unnormalized property names. I'm surprised the original implementation of adding this requested property didn't include making it `extensionAttribute.deploymentStatus`. I guess we have to keep the `Deployment status` name at this point, eh?

File: src/steps/devices/__snapshots__/index.test.ts.snap [hunk 2]
```diff
@@ -2175,6 +2175,7 @@ Object {
       "email": "erkang.zheng@lifeomic.com",
       "encrypted": true,
       "enrolledOn": 1591217314697,
+      "extensionAttribute.Deployment status": Array [],
```
Comment (aiwilliams) [inline src/steps/devices/__snapshots__/index.test.ts.snap:2178]: It's a shame our system isn't better at dropping `null`, `""`, `"   "`, and `[]` properties from the entities. We're shipping a ton of useless bytes for these empty values.

**Review by aiwilliams — DISMISSED** [review]

File: src/steps/devices/__snapshots__/index.test.ts.snap (hunk 2 above)
Comment (chasen-bettinger) [reply src/steps/devices/__snapshots__/index.test.ts.snap:2178]: FWIW, I wanted to make this property `undefined` by default, but I think this maintains existing functionality.

File: src/steps/devices/__snapshots__/index.test.ts.snap [hunk 3]
```diff
@@ -4351,6 +4352,7 @@ Object {
       "email": "erkang.zheng@lifeomic.com",
       "encrypted": true,
       "enrolledOn": 1555938179240,
+      "extensionAttribute.Deployment status": Array [],
```
Comment (erkangz) [inline src/steps/devices/__snapshots__/index.test.ts.snap:4355]: can we just normalize it to `deploymentStatus` instead?

**Review by erkangz — APPROVED** [review]

File: src/steps/devices/__snapshots__/index.test.ts.snap (hunk 3 above)
Comment (chasen-bettinger) [reply src/steps/devices/__snapshots__/index.test.ts.snap:4355]: That breaks backwards compatibility for query results and/or questions. Main focus is fixing a long-standing issue with NextRoll at the moment.

**Review by aiwilliams — APPROVED** [review]

**Pull request merged**